pkg/kubectl/resource_printer.go

```diff
@@ -2236,6 +2236,15 @@ func (h *HumanReadablePrinter) PrintObj(obj runtime.Object, output io.Writer) er
 		w = GetNewTabWriter(output)
 		defer w.Flush()
 	}
+
+	// check if the object is unstructured.  If so, let's attempt to convert it to a type we can understand before
+	// trying to print, since the printers are keyed by type.  This is extremely expensive.
+	if objBytes, err := runtime.Encode(api.Codecs.LegacyCodec(), obj); err == nil {
```

Comment:
can you not check whether this is unstructured first?  Or is it now *always* unstructured coming in to here?  Not sure why you have to blanket encode?